samples/extensions/dynamic_rendering/dynamic_rendering.cpp

```diff
@@ -353,8 +383,24 @@ void DynamicRendering::build_command_buffers()
 			draw_ui(draw_cmd_buffer);
 		};
 
+		VkImageSubresourceRange range{};
+		range.aspectMask     = VK_IMAGE_ASPECT_COLOR_BIT;
+		range.baseMipLevel   = 0;
+		range.levelCount     = VK_REMAINING_MIP_LEVELS;
+		range.baseArrayLayer = 0;
+		range.layerCount     = VK_REMAINING_ARRAY_LAYERS;
+
 		if (enable_dynamic)
 		{
+			vkb::insert_image_memory_barrier(draw_cmd_buffer,
+			                                 swapchain_buffers[i].image,
+			                                 0,
+			                                 VK_ACCESS_SHADER_WRITE_BIT,
+			                                 VK_IMAGE_LAYOUT_PRESENT_SRC_KHR,
+			                                 VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL,
+			                                 VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT,
```

Comment:
This must wait `pWaitDstStageMask` -- which would usually be `VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT`, but best to confirm that. :)